include/pybind11/pybind11.h

```diff
@@ -1913,8 +1913,7 @@ template <typename Type> class enum_ : public class_<Type> {
         def("__int__", [](Type value) { return (Scalar) value; });
         #if PY_MAJOR_VERSION < 3
             def("__long__", [](Type value) { return (Scalar) value; });
-        #endif
-        #if PY_MAJOR_VERSION > 3 || (PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 8)
+        #else
             def("__index__", [](Type value) { return (Scalar) value; });
```

Comment:
Isn't this function valid in >= 3.8? Or are we purposefully backporting index to early to python versions. We still will officially support 3.7

Reply:
Ahh, I misread the statement, this is a very confusing way to write it. Though I don't think this actually hurts older Pythons, it was just required for conversion in 3.8+.

Reply:
Actually, I prefer making this always present. `__index__` can be used other places, like in an index `x[y]` will call `y.__index__()`, even if it took till Python 3.8 to standardize that `__int__` meant lossy conversions, and `__index__` meant lossless conversions.